rsfbclient-native/src/connection.rs

```diff
@@ -588,6 +590,71 @@ impl<T: LinkageMarker> FirebirdClientSqlOps for NativeFbClient<T> {
     }
 }
 
+impl<T: LinkageMarker> FirebirdClientDbEvents for NativeFbClient<T> {
+    fn wait_for_event(
+        &mut self,
+        db_handle: &mut Self::DbHandle,
+        name: String,
+    ) -> Result<(), FbError> {
+        let mut event_buffer = ptr::null_mut();
+        let mut result_buffer = ptr::null_mut();
```

Comment:
It seems these buffers need to be freed with `isc_free` before the end of the function, as they are allocated by the `isc_event_block` function.
https://github.com/FirebirdSQL/php-firebird/blob/b6c288326678f7b8613f5a7d0648ad010c67674c/ibase_events.c#L45

Reply:
Solved